### PR TITLE
[BUGFIX LTS 3.28] Don't run getters while applying mixins

### PR DESCRIPTION
**Pull request by wycats**
Backports #20388

This change ensures that getters are never evaluated while applying
mixins.

It relies on the fact that all getters (including undecorated ones) get
converted into classic decorators when the mixin is originally created.

### CODE REVIEW

**Comment by wycats** [comment]
@kategengler I backported a bunch of the CI changes to get the tests passing but I'm not sure if I did it correctly. Can you give them a quick once over?

File: package.json [hunk 1]
```diff
@@ -150,7 +150,7 @@
     "tslint": "^5.20.1"
   },
   "engines": {
-    "node": "10.* || >= 12.*"
+    "node": ">= 14.*"
```
Comment (kategengler) [inline package.json:153]: We definitely cannot update engines in 3.28 point release

**Comment by kategengler** [comment]
@wycats What about CI was failing? We typically cherry-pick just commits back to the older versions and CI usually keeps working.

**Comment by wycats** [comment]
@kategengler I'll take another look at it. I think it might be tricky to avoid upgrades without somehow pinning other infrastructure like Embroider stuff, which we also might not want to do.

**Review by kategengler — APPROVED** [review]

**Pull request merged**